### PR TITLE
Add docker-compose overrides for SAML, LDAP, OAuth, and OIDC

### PR DESCRIPTION
**Pull request by chenbh**
The SAML one might be a bit weird to work with, but the others should be straight forward and mimics what we can expect from a real life scenario.

### CODE REVIEW

**Comment by chenbh** [comment]
Going to mark this as draft while we try to get some more auth connectors docker-compose'd

**Comment by chenbh** [comment]
@taylorsilva this is now ready for review

File: hack/overrides/oidc.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+version: '3'
+
+services:
+  web:
+    environment:
+      # CONCOURSE_MAIN_TEAM_OIDC_USER: user1
+      CONCOURSE_MAIN_TEAM_OIDC_USER: John
+      # CONCOURSE_MAIN_TEAM_OIDC_GROUP: group1
+
+      CONCOURSE_OIDC_ISSUER: http://oidc:9000
```
Comment (taylorsilva) [inline hack/overrides/oidc.yml:19]: A little bump in the dev experience here but not much we can do about it :(
Can we add a comment to the top explaining that when logging in with this override file they'll be directed to `http://oidc:9000` and they'll need to replace `oidc` with `localhost` to continue logging in. I can see someone that doesn't used docker often being confused by the behaviour

**Review by taylorsilva — CHANGES_REQUESTED** [review]
Everything worked perfectly! One nit

**Comment by chenbh** [comment]
Updated the comment, and also made the `./hack/dc` script not docker-compose up **EVERYTHING**

**Review by taylorsilva — APPROVED** [review]
lgtm

**Pull request merged**